Subscriber/OrderBackendSubscriber.php

```diff
@@ -156,6 +161,39 @@ public function onOrderPostDispatch(Enlight_Controller_ActionEventArgs $args)
         return $this->shipOrderToMollie($orderId);
     }
 
+    /**
+     * @param \Enlight_View_Default $view
+     * @param Enlight_Controller_Request_Request $request
+     */
+    private function extendOrderStatusHistory(\Enlight_View_Default $view, Enlight_Controller_Request_Request $request)
+    {
+        $orderId = $request->getParam('orderID');
+        $limit = $request->getParam('limit', 20);
+        $offset = $request->getParam('start', 0);
+        $sort = $request->getParam('sort', [['property' => 'history.changeDate', 'direction' => 'DESC']]);
+
+        /** @var Enlight_Components_Snippet_Namespace $namespace */
+        $namespace = Shopware()->Snippets()->getNamespace('backend/order');
+
+        //the backend order module have no function to create a new order so an order id must be passed.
+        if (empty($orderId)) {
+            $view->assign([
+                'success' => false,
+                'data' => $this->Request()->getParams(),
+                'message' => $namespace->get('no_order_id_passed', 'No valid order id passed.'),
+            ]);
+
+            return;
+        }
+
+        $history = $this->orderService->getOrderStatusHistory($orderId, $sort, $offset, $limit);
```

Comment:
short question, if we just extend the frontend, why do we need all of this?
just to add the "comments" column? is there another way besides having that bigger repo criteria query "duplicated"?

Reply:
It is required to get the content of the comments column.

We could load the comments via Ajax. Therefore we need an additional controller action and - because we should not load them one by one - some additional logic in ExtJs.